Process and send log messages

mtjandra opened the pull request:
* Send actual errors
* Allow user to provide additional context

dlockhart commented on src/logger.js:
```diff
@@ -4,13 +4,29 @@ export class Logger {
 		const logMessage = {};
+		logMessage.additionalContext = additionalContext || {};
+
+		if (error) {
```
There's no `error`, we probably just want to bail out?

mtjandra replied:
Hmm yeah I was thinking if they just had some custom message, but that doesn't seem to make much sense. Will update

dlockhart commented on src/logger.js:
```diff
@@ -4,13 +4,29 @@ export class Logger {
+			}
+			if (error.stack) {
+				logMessage.stackTrace = error.stack;
+			}
```
Any interest in the other [error properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) like `name`, `fileName`, `lineNumber` and `columnNumber`?

mtjandra replied:
So it seems chrome doesn't support `fileName`, `lineNumber` or `columnNumber`, but all this information is already captured in the stack trace including the error name. So I don't think it is necessary. 

It looks something like this

> TypeError: Cannot read property 'getFieldByName' of null
    at HTMLElement._handleSearch (http://127.0.0.1:8081/components/d2l-common/components/d2l-hm-search/d2l-hm-search.js:82:39)
    at HTMLElement._dispatchEvent (http://127.0.0.1:8081/components/d2l-inputs/d2l-input-search.js:221:10)
    at HTMLElement._handleInputKeyPress (http://127.0.0.1:8081/components/d2l-inputs/d2l-input-search.js:266:10)
    at HTMLInputElement.handler (http://127.0.0.1:8081/components/@polymer/polymer/lib/mixins/template-stamp.js:93:26)

dlockhart commented on src/logger.js:
```diff
@@ -4,14 +4,33 @@ export class Logger {
 
+		if (!error) {
+			return;
+		}
```
Super nit, but saves 3 lines: combine this with the `if (!this.loggingEndpoint || !error) {` line above.

dlockhart commented on src/logger.js:
```diff
@@ -4,14 +4,33 @@ export class Logger {
+			return;
+		}
+
+		if (!error.message && !error.stack) {
```
The `stack` being present below doesn't seem to set the `errorMessage`. Is checking `error.stack` here necessary?

mtjandra replied:
It was to handle this test case https://github.com/BrightspaceUI/logging/blob/b9f99da9d12f6ba896bd21001b8f1077e5228c83/test/logger.tests.js#L93

Perhaps I should update this to `typeof error === 'string'`

dbatiste commented:
A couple of small follow-up items to get ready for publishing:

* add the `files` sections to `package.json` to control which files are published (ex. [visual-diff](https://github.com/BrightspaceUI/visual-diff/blob/master/package.json#L16))
* add some readme info

Also, the eslint `node-config` isn't really quite right for this. It should probably be the `browser-config` for the logging module, but it looks like it needs to be updated for module support, including sort plugin, etc. Should be similar to the lit or polymer-3 configs, except without the html or lit plugins.  Once updated, the eslintrc.json here can probably just be:
```
{
  "extends": "brightspace/browser-config"
}
```

mtjandra commented:
>  It should probably be the browser-config ...

Okay I updated to use `brightspace/browser-config` and kept the additional rules not covered. Updated README and adding files section. Let me know if there is anything else needed to get ready for publishing. Any feedback on README or anything else is welcomed as well
